controllers/apim/virtualservice_controller.go

```diff
@@ -177,30 +170,13 @@ func normalizeStringMatch(sm *networkingv1alpha3.StringMatch) string {
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *VirtualServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	rlpMapper := &rateLimitPolicyMapper{
+		K8sClient: r.Client(),
+		Logger:    r.Logger(),
+	}
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&istionetworkingv1alpha3.VirtualService{}, builder.WithPredicates(virtualServicePredicate())).
+		For(&istionetworkingv1alpha3.VirtualService{}, builder.WithPredicates(routingPredicate(rlpMapper))).
```

Comment:
As an enhancement, instead of combining authorization and ratelimit logic into a single predicate, a cleaner approach would be to add multiple predicates (single responsability principle).

Reply:
it is good enough as it is now